src/nest.cpp

```diff
@@ -0,0 +1,24 @@
+#include <Rcpp.h>
+#include <dplyr.h>
+using namespace Rcpp;
+
+// [[Rcpp::export]]
+SEXP extract_groups(const List data, const CharacterVector nested_cols) {
+  ListOf<IntegerVector> indices(data.attr("indices"));
```

Comment:
I can't remember exactly when they get lost, but `do.grouped_df()` has:

```R
  # Force computation of indices
  if (is.null(attr(.data, "indices"))) {
    .data <- grouped_df_impl(.data, attr(.data, "vars"),
      attr(.data, "drop") %||% TRUE)
  }
```

dplyr should probably have an exported method to get the indices.

@romainfrancois can you remember when the grouped indices get dropped?